query/logicalplan/expr.go

```diff
@@ -255,13 +256,14 @@ func DynCol(name string) *DynamicColumn {
 	return &DynamicColumn{ColumnName: name}
 }
 
-func (c *DynamicColumn) DataType(s *dynparquet.Schema) (arrow.DataType, error) {
-	colDef, found := s.ColumnByName(c.ColumnName)
-	if !found {
-		return nil, errors.New("column not found")
+func (c *DynamicColumn) DataType(s *parquet.Schema) (arrow.DataType, error) {
```

Comment:
This doesn’t seem quite right. We won’t find an exact match, but we should still be able to use the parquet schema by just checking the prefix.

I’m not sure though what will happen if no column exists for a dynamic column, which can happen. Can we add a test for that?

Reply:
I fixed it to correctly match the dynamic schema columns. However, I don't think this code is actually reachable currently. We only call this in `pqarrow/arrow.go:72` which only happens if the distinct expr is computed which `DynamicColumn` isn't.